QEfficient/utils/constants.py

```diff
@@ -10,7 +10,41 @@
 UTILS_DIR = os.path.dirname(os.path.abspath(__file__))
 QEFF_DIR = os.path.dirname(UTILS_DIR)
 ROOT_DIR = os.path.dirname(QEFF_DIR)
-QEFF_MODELS_DIR = os.path.join(ROOT_DIR, "qeff_models")
+
+
+# Store the qeff_models inside the ~/.cache directory or over-ride with an env variable.
+def get_models_dir():
+    """
+    Determine the directory for storing QEFF models.
+
+    Priority:
+    1. Use $XDG_CACHE_HOME/qeff_models if XDG_CACHE_HOME is set.
+    2. Use QEFF_HOME if set in environment.
+    3. Default to ~/.cache/qeff_models.
+
+    Sets QEFF_MODELS_DIR environment variable if not already set.
+
+    Returns:
+        str: Path to the QEFF models directory.
+    """
+    qeff_cache_home = os.environ.get("QEFF_HOME")
+    # Check if XDG_CACHE_HOME is set
+    xdg_cache_home = os.environ.get("XDG_CACHE_HOME")
+    if xdg_cache_home:

```

Comment:
It should be other way around, first check for QEFF_HOME, only if not present you have to check for XDG_CACHE_HOME.

Reply:
Done